horizon/facts/client.py

```diff
@@ -57,7 +60,7 @@ async def build_forward_request(
         return self.client.build_request(
             method=request.method,
             url=full_path,
-            params=request.query_params,
+            params={**request.query_params, **(query_params or {})},
```

Comment:
I would merge it first in a variable then use it. it's would be easier to understand and debug

Reply:
fixed